src/main/kotlin/nl/avisi/structurizr/site/generatr/GenerateSiteCommand.kt

```diff
@@ -73,8 +82,31 @@ class GenerateSiteCommand : Subcommand(
             refreshLocalClone()
         }
 
-        val branchNames = branches.split(",")
+        val branchNames: MutableList<String> = if (allBranches) {
+            clonedRepository.getBranchNames(excludeBranches.split(",")).toMutableList()
+        } else {
+            branches.split(",").toMutableList()
+        }
+
+        println("Branches : $branchNames")
+
         val workspaceFileInRepo = File(clonedRepository.cloneDir, workspaceFile)
+        val errorBranches: MutableList<String> = mutableListOf()
+
+        branchNames.forEach { branch ->
+            try {
+                println("Checking branch $branch")
+                clonedRepository.checkoutBranch(branch)
+                createStructurizrWorkspace(workspaceFileInRepo)
+            } catch (e: Exception) {
+                errorBranches.add(branch)
+            }
+        }
+
+        println("Branches with workspace errors : $errorBranches")
+        branchNames.removeAll(errorBranches)
+
+        println("Generating sites for the following branches : $branchNames")
```

Comment:
This can also be done without mutable lists, to make the code more idiomatic and concise. After applying this suggestion, the code after this needs to be updated to use `branchesToGenerate` instead of `branchNames`. I couldn't add that to the suggestion, so you'll have to do that manually.

```suggestion
        val branchNames = if (allBranches)
            clonedRepository.getBranchNames(excludeBranches.split(","))
        else
            branches.split(",")

        val workspaceFileInRepo = File(clonedRepository.cloneDir, workspaceFile)
        val branchesToGenerate = branchNames.filter { branch ->
            println("Checking branch $branch")
            clonedRepository.checkoutBranch(branch)
            try {
                createStructurizrWorkspace(workspaceFileInRepo)
                true
            } catch (e: Exception) {
                false
            }
        }

        println("The following branches contain a valid Structurizr workspace: $branchesToGenerate")
```